fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4

```diff
@@ -50,7 +50,7 @@ statement
         (ROLLUP LEFT_PAREN rollupDefs RIGHT_PAREN)?
         propertyClause?
         (AS query)?                                                    #createTable
-    | explain? INSERT (INTO | OVERWRITE TABLE)
+    | explain? INSERT selectHint? (INTO | OVERWRITE TABLE)
```

Comment:
rename it to hint since we use it not just in select statement

Reply:
there's already a `class Hint`, so I rename it to statementhint